src/evaluation_system/tests/plugin_command_test.py

```diff
@@ -131,18 +132,18 @@ def test_plugin_status(dummy_env, caplog) -> None:
 
     import freva
 
-    res = freva.run_plugin("dummyplugin", the_number=2, other=-5, batchmode=True)
-    with pytest.raises(ValueError):
-        res.wait(2)
-    assert res.status == "running"
-    res.kill()
-    time.sleep(0.5)
-    assert res.status == "broken"
-    res.kill()
-    res = freva.run_plugin("dummyplugin", the_number=2, other=-1, batchmode=True)
+    # res = freva.run_plugin("dummyplugin", the_number=2, other=-5, batchmode=True)
+    # # with pytest.raises(ValueError):
+    # # res.wait()
+    # # assert res.status == "running"
+    # res.kill()
+    # time.sleep(0.5)
+    # assert res.status == "broken"
+    # res.kill()
+    res = freva.run_plugin("dummyplugin", the_number=2, other=-1, batchmode=False)
```

Comment:
Yes, but I think there is problem with this `batchmode=True` in unit-test! we don't get any other labels than `scheduled`  to testify them when we use `batchmode=True`. It was kinda shortcut solution, not the fundamental.

Reply:
@antarcticrainforest due to the problem that I've mentioned, we get failure test on `test_plugin_status`. Do you habe any idea for this? Do i need to comment it for the current state or you will take over?

Reply:
Have you tried this on your computer?

Reply:
btw docs service got canceled. i don't know why it took too long and finally red!

Reply:
I ran this test, and saw the same error message, a second time went through. If I do this by hand then it works. I am not sure what is going on here.  I any case we do want theses tests, they are important. The only thing is, we have to figure out why this doesn't work at times.

Reply:
Let me finish crawler, then I will deep diver here to find out the roots of error ....

Reply:
Sorry, I didn't see your message. I think could reproduce the error by setting up a new conda env with the latest python. Something must have changed the way things get imported during the update from python3.11 to python3.12.  The reason why the plugin is stuck in scheduled is:

```
home/wilfred/workspace/freva/src/evaluation_system/misc/config.py:166: Warning: /home/wilfred/.anaconda3/envs/freva-dev/freva/freva.crt not found. Secrets are stored in central vault and apublic key is needed to open the vault. Without the public key you won't be probably be able to establish as database connection.
  warnings.warn(
[13:26:19] ERROR    freva - ERROR - 'NoneType' object has no         utils.py:98
                    attribute 'startswith' - increase verbosity                 
                    flags (-v) for more information                             
                    ╭───── Traceback (most recent call last) ──────╮            
                    │ /home/wilfred/.anaconda3/envs/freva-dev/lib/ │            
                    │ python3.12/site-packages/django/db/backends/ │            
                    │ mysql/base.py:227 in get_connection_params   │            
                    │                                              │            
                    │   224 │   │   │   kwargs["database"] = setti │            
                    │   225 │   │   if settings_dict["PASSWORD"]:  │            
                    │   226 │   │   │   kwargs["password"] = setti │            
                    │ ❱ 227 │   │   if settings_dict["HOST"].start │            
                    │   228 │   │   │   kwargs["unix_socket"] = se │            
                    │   229 │   │   elif settings_dict["HOST"]:    │            
                    │   230 │   │   │   kwargs["host"] = settings_ │            
                    ╰──────────────────────────────────────────────╯            
                    AttributeError: 'NoneType' object has no                    
                    attribute 'startswith'   

```

And the reason why that happens is, that the wrong evaluation_system file is passed to the workload manager.

To fix this you can change the `_ConfigWrapper` class in `/src/evaluation_system/misc/__init__.py` to

```
lass _ConfigWrapper:
    """Convenience class that helps to dynamically set the location of the
    evaluation system config file."""

    _env: str = "EVALUATION_SYSTEM_CONFIG_FILE"

    def __init__(
        self,
        default_file: str,
    ):
        self.default_file = default_file

    def __str__(self) -> str:
        return os.environ.get(self._env, self.default_file)

    def __repr__(self) -> str:
        return self.__str__()

    def __fspath__(self) -> str:
        return os.environ.get(self._env, self.default_file)
```

I'll see if I can submit a proposal you can apply in the review. 

Also, pleas put back all the tests related to wait. Same goes for the doc strings.

Reply:
I couldn't add a suggestion to the code because you didn't touch this part. So if you simply delete the `(str)` from the class inheritance and add the `__fspath__` method you should be good to go.